torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3580,7 +3633,10 @@ def _cast_grad_to_param_dtype(
     def _should_free_unsharded_flat_param(self, handle: FlatParamHandle):
```

Comment:
Before this PR: I wonder if we should add a comment that this method is meant for the post-backward reshard only.